src/undate/undate.py

```diff
@@ -456,20 +457,48 @@ def duration(self) -> Timedelta:
         # calculate month duration within a single year (not min/max)
         if self.precision == DatePrecision.MONTH:
             latest = self.latest
+            # if year is unknown, calculate month duration in
+            # leap year and non-leap year, in case length varies
             if not self.known_year:
-                # if year is unknown, calculate month duration in
-                # a single year
-                latest = Date(self.earliest.year, self.latest.month, self.latest.day)
+                # TODO: should leap-year specific logic shift to the calendars,
+                # since it works differently depending on the calendar?
+                possible_years = [
+                    self.calendar_converter.LEAP_YEAR,
+                    self.calendar_converter.NON_LEAP_YEAR,
+                ]
+            # TODO: what about partially known years like 191X ?
+            else:
+                # otherwise, get possible durations for all possible months
+                # for a known year
+                possible_years = [self.earliest.year]
+
+            # for every possible month and year, get max days for that month,
+            possible_max_days = set()
+            # appease mypy, which says month values could be None here
+            if self.earliest.month is not None and self.latest.month is not None:
+                for possible_month in range(self.earliest.month, self.latest.month + 1):
+                    for year in possible_years:
+                        possible_max_days.add(
+                            self.calendar_converter.max_day(year, possible_month)
+                        )
+
+                # if there is more than one possible value for month length,
+                # whether due to leap year / non-leap year or ambiguous month,
+                # return an uncertain delta
+                if len(possible_max_days) > 1:
+                    return UnDelta(*possible_max_days)
+
+            # otherwise, calculate timedelta normally
+            max_day = list(possible_max_days)[0]
```

Comment:
in case that `if self.earliest.month is not None and self.latest.month is not None:` is not true, I think this here would throw an list index out of range error?

Reply:
thanks for catching; I wonder if I got the nesting wrong... I'll try to figure out a unit test case to trigger that error

Reply:
@jdamerow tl;dr is that I got the nesting wrong. 

The `Date` object allows month to be None, but undate earliest/latest are always calculated as full-precision dates.  Not sure if there's something we can do to improve this ambiguity on the Date object + precision.

I added some notes about ambiguous year duration, but want to handle that in a separate branch - it's tricky because it depends on the calendar!